mne/simulation/source.py

```diff
@@ -312,3 +317,188 @@ def simulate_stc(src, labels, stc_data, tmin, tstep, value_fun=None,
+        if not isinstance(label, Label):
+            raise ValueError('label must be a Label,'
+                             'not %s' % type(label))
+
+        # If it is not a list then make it one
+        if not isinstance(waveform, list) or len(waveform) == 1:
+            waveform = [waveform] * len(events)
+            # The length is either equal to the length of events, or 1
+        if len(waveform) != len(events):
+            raise ValueError('Number of waveforms and events should match '
+                             'or there should be a single waveform')
+        # Update the maximum duration possible based on the events
+        self._labels.extend([label] * len(events))
+        self._waveforms.extend(waveform)
+        self._events = np.vstack([self._events, events])
+        # First sample per waveform is the first column of events
+        # Last is computed below
+        self._last_samples = np.array([self._events[i, 0] + len(w)
+                                      for i, w in enumerate(self._waveforms)])
+
+    def get_stim_channel(self, start_sample=0, nb_samples=None):
+        """Get the stim channel from the provided data.
+
+        Returns the stim channel data according to the simulation parameters
+        which should be added through function add_data. If both start_sample
+        and nb_samples are not specified, the entire duration is used.
+
+        Parameters
+        ----------
+        start_sample : int
+            First sample in chunk. Default is 0.
+        nb_samples : int
+            Number of samples in the stc data. If both start_sample and
+            nb_samples are not specified, the entire duration is used.
+
+        Returns
+        -------
+        stim_data : 1-d array of int
+            The stimulation channel data.
+        """
+        if nb_samples is None:
+            nb_samples = self.nb_samples - start_sample
+
+        end_sample = start_sample + nb_samples
+        # Initialize the stim data array
+        stim_data = np.zeros(nb_samples, dtype=int)
+
+        # Select only events in the time chunk
+        stim_ind = np.where(np.logical_and(self._events[:, 0] >= start_sample,
+                                           self._events[:, 0] < end_sample))[0]
+
+        if len(stim_ind) > 0:
+            relative_ind = self._events[stim_ind, 0].astype(int) - start_sample
+            stim_data[relative_ind] = self._events[stim_ind, 2]
+
+        return stim_data
+
+    def get_stc(self, start_sample=0, nb_samples=None):
+        """Simulate a SourceEstimate from the provided data.
+
+        Returns a SourceEstimate object constructed according to the simulation
+        parameters which should be added through function add_data. If both
+        start_sample and nb_samples are not specified, the entire duration is
+        used.
+
+        Parameters
+        ----------
+        start_sample : int
+            First sample in chunk. Default is 0.
+        nb_samples : int
```

Comment:
int | None

Reply:
It cannot be `None`, but it is optional.

Reply:
You are right, I was looking at the wrong line.